Sources/ResilientDecoding/ErrorReporting.swift

```diff
@@ -1,43 +1,114 @@
-// Created by George Leontiev on 3/28/20.
+// Created by George Leontiev on 3/25/20.
 // Copyright © 2020 Airbnb Inc. All rights reserved.
 
 import Foundation
 
-// MARK: - Error Reporter
+// MARK: - Enabling Error Reporting
+
+extension Dictionary where Key == CodingUserInfoKey, Value == Any {
+
+  /**
+   Creates and registers a `ResilientDecodingErrorReporter` with this `userInfo` dictionary. Any `Resilient` properties which are decoded by a `Decoder` with this user info will report their errors to the returned error reporter.
+   - note: May only be called once on a particular `userInfo` dictionary
+   */
+  public mutating func enableResilientDecodingErrorReporting() -> ResilientDecodingErrorReporter {
+    if let existingValue = self[resilientDecodingErrorReporterCodingUserInfoKey] {
+      assertionFailure()
+      if let existingReporter = existingValue as? ResilientDecodingErrorReporter {
+        existingReporter.currentDigest.mayBeMissingReportedErrors = true
+      }
```

Comment:
Can you help me understand this part? Why do we set `mayBeMissingReportedErrors = true`?

Reply:
I realize this was preexisting.

Reply:
Ah I see below. Because we're recreating this we may have lost errors. Nice.